air/src/trace/main_trace.rs

```diff
@@ -367,6 +357,30 @@ impl MainTrace {
         self.columns.get_column(MEMORY_V_COL_RANGE.start + 3)[i]
     }
 
+    /// Computes a memory read request at `row` given randomness `alphas`, memory address `addr`
+    /// and value `value`.
+    pub fn compute_memory_read_request<E: FieldElement<BaseField = Felt>>(
```

Comment:
With this new function, we have 2 functions in the system that build a memory read request: this one, and `processor::chiplets::aux_trace::build_mem_request()`. We should have only one.

A few thoughts:

- I think `MainTrace` should only have dumb accessors. It shouldn't be computing complex functions of the trace columns.
- A memory read (for words) is always a random linear combination of `[op_label, ctx, addr, clk, VALUE]`. The issue we have is that `MLOAD*` and `MSTORE*` don't take these values from the same locations in the trace as `RCOMBBASE`. The solution IMO is to have `build_mem_request` take the values it needs as input, and caller can supply them by reading from the correct trace location.
- I would split `build_mem_request()` into 2: one that builds a memory access for a `Felt` (e.g. `build_mem_element_request()`), and another for a `Word` (e.g. `build_mem_word_request()`). This would help us think of them as separate requests, because they are (i.e. the virtual columns that they collapse into 1 are different).

In other words, we currently think of a memory request (as defined by `build_mem_request()`) as coming from `MLOAD*` or `MSTORE*`. This change would help us think of a memory request as a random linear combination of specific columns (depending on whether we're reading a `Felt` or `Word`), which can come from any operation (current and/or future ones).

Reply:
> With this new function, we have 2 functions in the system that build a memory read request: this one, and `processor::chiplets::aux_trace::build_mem_request()`.

Ah - good point! I forgot about this one. We actually also have `build_mstream_request()` which does something similar. I'd love to consolidate them to use the same logic for computing the random linear combination.

> I think `MainTrace` should only have dumb accessors. It shouldn't be computing complex functions of the trace columns.

I agree with this. I wanted to have a more "convenient" location for the function - but separation of concerns is probably more important here.

As mentioned above, I still think that it would be good to have the function that computes the random linear combination for the request - but this function could probably live in the `aux_trace` module. Than, this function could be used from `build_mem_request()`, `build_mstream_request()`, `build_rcomb_base_request()` and others in the future.

> I would split `build_mem_request()` into 2: one that builds a memory access for a `Felt` (e.g. `build_mem_element_request()`), and another for a `Word` (e.g. `build_mem_word_request()`). This would help us think of them as separate requests, because they are (i.e. the virtual columns that they collapse into 1 are different).

If we have the above-mentioned function as a helper, I think splitting up `build_mem_request()` would make sense.

Reply:
I agree with points 1 and 2. Regarding point 3, it is indeed helpful (especially for the future) to make the code reflect the separation between requests (and by transitivity virtual columns). The issue is that this can be in conflict with reducing the size of the code in some places. Nevertheless, I have split the `build_mem_request()` and we can discuss the other cases, if any, on a case-by-case basis